### PR TITLE
tests/dns: add tests for task 7018 - v2

### PR DESCRIPTION
**Pull request by jufajardini**
Previous PR: https://github.com/OISF/suricata-verify/pull/1897/files

Describe changes: 
- differentiate `flow` checks for 6.0.x, since `tcp.*_max_regions` are from 7.0.x onwards
- rebase

## Ticket

Redmine ticket:
https://redmine.openinfosecfoundation.org/issues/7018

### CODE REVIEW

**Comment by jufajardini** [comment]
Suricata PR already merged xD

File: tests/dns/task-7018-dns-ids-stream-rule/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+The capture shows three request-response DNS transactions:
+Query 1: suricata.io
+Query 2: oisf.net
+Query 3: suricata.org
+
+We match those against a single payload rule without any DNS keywords,
+and inspecting content `suricata|02|`. Ideally, the expectation is to have 2 alerts,
+for the portion of the stream associated with Query 1 - that's because on the
+wire we observe that for Query three the content is `suricata|03|`.
```
Comment (jasonish) [inline tests/dns/task-7018-dns-ids-stream-rule/README.md:20]: Being a test to prove expected behavior, can we drop the word "Ideally"?

File: tests/dns/task-7018-dns-ids-stream-rule/test.yaml [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+      pcap_cnt: 7
+      alert.signature: DNS suricata query - payload rule
+      alert.signature_id: 1
+      proto: TCP
+      src_ip: 10.16.1.11
+      src_port: 36926
+      app_proto: dns
+      dest_ip: 9.9.9.9
+      dest_port: 53
+      direction: to_server
```
Comment (jasonish) [inline tests/dns/task-7018-dns-ids-stream-rule/test.yaml:26]: Should we add transaction data here? Suricata commit bb45ac71ef572acb5591c135eb3c73e901a1cc51 changes which transaction gets logged here, which I think is an important deal of this test.

File: tests/dns/task-7018-dns-ips-stream-rule/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+## Behavior
+
+We match those against a single payload rule without any DNS keywords,
+and inspecting content `suricata|02|`. Ideally, the expectation is to have 2 alerts,
+for the portion of the stream associated with Query 1 - that's because on the
+wire we observe that for Query three the content is `suricata|03|`.
+
+For IPS mode, as a larger portion of the stream buffer is kept available, and as
+it still contains the matching bytes, we'll see more alerts triggered.
+
```
Comment (jasonish) [inline tests/dns/task-7018-dns-ips-stream-rule/README.md:24]: I think we should also call-out the undesirable behavior here.  Our content is looking for `suricata|02|`, however we are logging DNS `alerts` with transaction data for `oisf.net`.

Maybe also test for that undesirable behavior as well.

**Review by jasonish — CHANGES_REQUESTED** [review]
A few changes required IMO to better reflect what we're testing.

File: tests/dns/task-7018-dns-ids-stream-rule/test.yaml (hunk 2 above)
Comment (jufajardini) [reply tests/dns/task-7018-dns-ids-stream-rule/test.yaml:26]: Will do. Wasn't sure what would be the final decision on logging tx data for stream rules, so ended up not having that in the checks, but will bring them back.

**Comment by jufajardini** [comment]
Feedback finally incorporated with https://github.com/OISF/suricata-verify/pull/1967

**Pull request closed without merging**